python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+
+    @pipeBase.timeMethod
+    def ingest(self, cat, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog passed as an object.
+
+        @param cat (BaseCatalog) Catalog to ingest.
+        @param tableName (str)   Name of the database table to create.
+        @param host (str)        Name of the database host machine.
+        @param db (str)          Name of the database to ingest into.
+        @param port (int)        Port number on the database host.
+        @param user (str)        User name to use for the database.
+        @param viewName (str)    Name of the database view to create.
+        """
+        tableName = quoteIdentifier(tableName)
+        viewName = quoteIdentifier(viewName) if viewName else None
+        with closing(self.connect(host, port, db, user)) as conn:
+            # Determine the maximum query length (MySQL-specific) if not
+            # configured.
+            if self.config.maxQueryLen is None:
+                with closing(conn.cursor()) as cursor:
+                    cursor.execute(
+                        """SELECT variable_value
+                        FROM information_schema.session_variables
+                        WHERE variable_name = 'max_allowed_packet'
+                        """
+                    )
+                    maxQueryLen = int(cursor.fetchone()[0])
+            else:
+                maxQueryLen = self.config.maxQueryLen
+            self.log.logdebug("maxQueryLen: {}".format(maxQueryLen))
+            self._createTable(conn, tableName, cat.schema)
+            if viewName is not None:
+                self._createView(conn, tableName, viewName, cat.schema)
+            self._ingest(conn, cat, tableName, maxQueryLen)
+
+    @staticmethod
+    def connect(host, port, db, user=None):
+        # Map BIT columns to Python bools
+        conv = dict(MySQLdb.converters.conversions)
+        conv[MySQLdb.constants.FIELD_TYPE.BIT] = lambda v: v != '\x00'
+        kwargs = dict(host=host, port=port, db=db, conv=conv)
+        if user is not None:
+            kwargs["user"] = user
+        try:
+            # See if we can connect without a password (e.g. via my.cnf)
+            return MySQLdb.connect(**kwargs)
+        except:
+            # Fallback to DbAuth
+            kwargs["user"] = DbAuth.username(host, str(port))
+            kwargs["password"] = DbAuth.password(host, str(port))
+            return MySQLdb.connect(**kwargs)
+
+    def _executeSql(self, conn, sql):
+        """Execute a SQL query with no expectation of a result.
+        """
+        self.log.logdebug(sql)
+        conn.query(sql)
+
+    def _schemaItems(self, schema):
+        """A generator over ingestible schema items.
+        """
+        for item in schema:
+            field = item.field
+            if field.getTypeString() not in fieldFormatters:
+                self.log.warn(
+                    "Skipping field {}: type {} not supported".format(
```

Comment:
I don't think we actually have a lazy formatting interface in the `lsst.pex.logging` Python interface.  The proper invocation for lazy logging in the current `lsst.log` interface is `self.log.warn("skipping field %s: %s not supported", field.getName(), field.getTypeString())`.  But as Serge says, lazy formatting is completely unnecessary here since it's a warning and not performance-critical.  And if we're not using lazy formatting, `.format()` is preferred to `%`.